tests/oauth/test_exchange_token.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from respx import MockRouter
+
+from spylib.oauth import OfflineTokenModel, OnlineTokenModel, exchange_token
+
+exchange_token_params = [
+    (
+        dict(
+            access_token='offline-token',
+            scope='read_products',
+        ),
+        OfflineTokenModel,
+    ),
+    (
+        dict(
+            access_token='online-token',
+            scope='write_products',
+            expires_in=86399,
+            associated_user_scope='read_products',
+            associated_user=dict(
+                id=902541635,
+                first_name='John',
+                last_name='Smith',
+                email='john@example.com',
+                email_verified=True,
+                account_owner=True,
+                locale='en',
+                collaborator=False,
+            ),
+        ),
+        OnlineTokenModel,
+    ),
+]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('token_dict,token_model', exchange_token_params)
+async def test_exchange_token(respx_mock: MockRouter, token_dict, token_model):
+    shop = 'example.myshopify.com'
+    code = 'code'
+    api_key = 'api-key'
+    api_secret_key = 'api-secret-key'
+
+    respx_mock.post(
+        f'https://{shop}/admin/oauth/access_token',
+        json__code=code,
+        json__client_id=api_key,
+        json__client_secret=api_secret_key,
+    ).respond(
+        json=token_dict,
+    )
+
+    result = await exchange_token(
+        shop=shop,
+        code=code,
+        api_key=api_key,
+        api_secret_key=api_secret_key,
+    )
+
+    assert result == token_model.parse_obj(token_dict)
```

Comment:
I think this needs to be tested manually because both dicts go through the same model processing.
```suggestion
    assert result == token_model.parse_obj(token_dict)
    parsed_scopes = token_dict['scope'].split(',')
    assert result.scopes == parsed_scopes
    if 'associated_user_scope' in token_dict:
        parsed_user_scopes = token_dict['associated_user_scope'].split(',')
        assert result.associated_user_scopes == parsed_user_scopes
```

Reply:
> I think this needs to be tested manually because both dicts go through the same model processing.

I don't really understand...do you mean in order to test the models? I can write parsing tests for those elsewhere if that is a concern

Reply:
yea, what do you mean "both dicts go through the same model processing"? `token_model` is of type `OfflineTokenModel` in first test and `OnlineTokenModel` in the second test.

Reply:
I mean if you screw up the scopes parsing, your test only won't catch it since the same dict will go through the model. It's just to test your validator essentially. One could make the argument that it should be done separately though thinking about it.